http.c

```diff
@@ -1726,6 +1751,22 @@ evhttp_parse_request_line(struct evhttp_request *req, char *line)
 	} /* switch */
 
 	if ((int)type == EVHTTP_REQ_UNKNOWN_) {
+		/* check extended methods */
+		const struct evhttp_extended_method * ext_methods = req->evcon->ext_methods;
+		if (ext_methods == NULL && req->evcon->http_server != NULL)
+			ext_methods = req->evcon->http_server->ext_methods;
+		if (ext_methods != NULL) {
+			for (; ext_methods->method != NULL; ext_methods++) {
+				if (method_len == strlen(ext_methods->method) &&
+				    0 == memcmp(method, ext_methods->method, method_len)) {
```

Comment:
oh man. couldn't we just do pointer comparisons if the user is defining the const?